src/main/java/ca/corefacility/bioinformatics/irida/ria/web/users/UsersAjaxController.java

```diff
@@ -51,8 +54,59 @@ public UsersAjaxController(UIUsersService UIUsersService) {
 	 * @return {@link ResponseEntity} internationalized response to the update
 	 */
 	@RequestMapping("/edit")
+	@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_MANAGER')")
 	public ResponseEntity<String> updateUserStatus(@RequestParam Long id, @RequestParam boolean isEnabled,
 			Locale locale) {
 		return UIUsersService.updateUserStatus(id, isEnabled, locale);
 	}
+
+	/**
+	 * Submit a user edit
+	 *
+	 * @param userId      The id of the user to edit (required)
+	 * @param firstName   The firstname to update
+	 * @param lastName    the lastname to update
+	 * @param email       the email to update
+	 * @param phoneNumber the phone number to update
+	 * @param systemRole  the role to update
+	 * @param userLocale  The locale the user selected
+	 * @param enabled     whether the user account should be enabled or disabled.
+	 * @param principal   a reference to the logged in user.
+	 * @param request     the request
+	 * @return The name of the user view
+	 */
+	@RequestMapping(value = "/{userId}/edit", method = RequestMethod.POST)
+	public ResponseEntity<Map<String, String>> updateUser(@PathVariable Long userId,
+			@RequestParam(required = false) String firstName, @RequestParam(required = false) String lastName,
+			@RequestParam(required = false) String email, @RequestParam(required = false) String phoneNumber,
+			@RequestParam(required = false) String systemRole,
+			@RequestParam(required = false, name = "locale") String userLocale,
+			@RequestParam(required = false) String enabled, Principal principal, HttpServletRequest request) {
+
+		UserDetailsResponse response = UIUsersService.updateUser(userId, firstName, lastName, email, phoneNumber,
+				systemRole, userLocale, enabled, principal, request);
+
+		if (response.hasErrors())
+			return ResponseEntity.status(HttpStatus.CONFLICT)
```

Comment:
Change `HttpStatus.CONFLICT` to `HttpStatus.BAD_REQUEST` since not all the errors are conflicts